bionlp/evaluate/evaluation.py

```diff
@@ -65,23 +63,23 @@ def get_Approx_Metrics(y_true, y_pred, verbose=True, preMsg='', flat_list=False)
     return avg_f1
 
 
-def get_ConfusionMatrix(true, predicted, final_eval=False):
+def get_ConfusionMatrix(true, predicted, final_eval=False, final_eval_out_file=''):
     # Confusion Matrix is only valid for partial evaluation.
     true_chain = list(itertools.chain.from_iterable(true))
     predicted_chain = list(itertools.chain.from_iterable(predicted))
     msg = "Confusion Matrix of combined folds (partial evaluation)\n{0}".format(
         ConfusionMatrix(true_chain, predicted_chain))
     print(msg)
-    if final_eval and final_eval_out_file:
-        append_message_to_final_eval_file(msg)
+    if final_eval and final_eval_out_file is not "":
```

Comment:
`final_eval` is a Boolean variable that tells if the current evaluation is the final one
`final_eval_out_file` is the file into which the results of the final evaluation are to be saved.

I will change the name of `final_eval` to `is_final_eval` to make this more obvious.